src/Microsoft.OData.Core/PropertyAndAnnotationCollector.cs

```diff
@@ -234,7 +232,7 @@ internal void Reset()
         internal void AddODataScopeAnnotation(string annotationName, object annotationValue)
         {
             Debug.Assert(!string.IsNullOrEmpty(annotationName));
-            Debug.Assert(JsonLight.ODataJsonLightReaderUtils.IsODataAnnotationName(annotationName));
+            Debug.Assert(Json.ODataJsonReaderUtils.IsODataAnnotationName(annotationName));

```

Comment:
```suggestion
            Debug.Assert(ODataJsonReaderUtils.IsODataAnnotationName(annotationName));
```
You already have a `Microsoft.OData.Json` using.